### DISCUSSION TITLE
6.3.2에 나와있는 설명이 이해가 가지 않습니다.

### DISCUSSION THREAD

**Discussion by seonkyungkim** [fsoftwareengineer/todo-application#55] [질문과 답변] [answered]
안녕하세요, 작가님.

저는 eBook으로 책을 읽고 있는데요. 아래 사진과 같이 p.355의 첫번째 문장을 잘못 이해하고 실습을 진행한 것 같아 질문드립니다.

![image](https://user-images.githubusercontent.com/57647921/148900731-bd9c8075-aab5-41eb-83a3-aec5d6a07279.png)

1. 'EB CLI를 이용해 로컬 환경에서 애플리케이션을 생성한 후'의 의미에 대해 한참을 생각해봐도 이해가 가지 않아 건너뛴 뒤에
2. 터미널에서 지금까지 작성한 백엔드 프로젝트 todoapp이 있는 디렉터리로 이동하고 ($ cd IdeaProjects)
3. $ eb init todoapp 명령어로 초기화를 진행했습니다.

그 결과 .elasticbeanstalk/config.yml 파일의 profile이 책의 예제 6-8처럼 eb-cli로 나오는 대신 null로 등록되어 있었으며, 무시하고 6.3.4의 eb create 실습을 진행하자 jar 경로가 잘못되었다는 오류가 출력되었습니다.

첫번째 문장을 건너뛴 게 잘못인 것 같아서 config.yml 파일의 profile을 임의로 eb-cli로 수정해보았으나,
다시 eb init으로 초기화를 진행하라는 에러메시지만 출력될 뿐 정상적으로 동작하지 않았습니다.

EB CLI를 이용해 로컬 환경에서 애플리케이션을 어떻게 생성해야 하는지, 어떤 의미인지 자세하게 설명해주시면 감사하겠습니다.

**Comment by fsoftwareengineer** [comment, marked as answer]
EB CLI를 이용해 로컬 환경에서 어플리케이션을 생성한 후 -> 생성하기 위해

입니다.

**Reply by seonkyungkim** [reply]
아~ 그럼 그 부분은 결과적으로 건너뛰는게 맞다는 말씀이시죠?ㅎㅎ 감사합니다. 근데 그럼 이후에 발생한 오류들은 원인이 뭘까요... 
config.yml에서 profile이 eb-cli로 나오지 않고 null로 나와있었던 것이나 eb create로 환경설정이 되지 않았던 문제도 도움 부탁드립니다.

**Comment by fsoftwareengineer** [comment]
에러가 나는 이유는 프로젝트의 루트 디렉터리가 아니라서 그런걸까요? 보통 IdeaProjects안에 다른 인텔리제이 프로젝트들이 있어서 여쭤봅니다. 만약 그렇다면 프로젝트 루트 디렉터리인 demo(책에서) 또는 혹시 프로젝트 이름이 todoapp이라면 todoapp내부에서 eb init을 진행해야 합니다.


cd todoapp
eb init todoapp

**Reply by seonkyungkim** [reply]
친절하고 빠른 답변 정말 감사합니다 작가님!!

**Reply by wondongkyu** [reply]
config.yml에서 profile이 eb-cli로 나오지 않는 이유는 무엇인가요 책에 나와있는데로 했는데도 계속 null로 찍히네요

**Reply by wondongkyu** [reply]
문제 해결했습니다~
